tests/modules/test_cloud.py

```diff
@@ -39,50 +39,104 @@ def assert_tensor_list_diff(a, b):
+                        cfg.rsync_up(tmp_path, rand_folder)
+
+    
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",return_value={"gs://iordanis/bucket1","gs://iordanis/bucket2",f"gs://{Path(cfg.bucket) / rand_folder}/"}):
+        new_files = cfg.list_bucket()
     rand_destination = f"gs://{Path(cfg.bucket) / rand_folder}/"
     assert set(new_files).difference(files) == {rand_destination}
-    uploaded_files = cfg.list_bucket(rand_folder + "/" + tmp_path.name)
+    uploaded_files=None
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.list_bucket",return_value={f"{rand_destination}/{tmp_path.name}/tenosr1",f"{rand_destination}/{tmp_path.name}//tenosr2"}):
+        uploaded_files = cfg.list_bucket(rand_folder + "/" + tmp_path.name)
     assert len(uploaded_files) == 2
     # Replace original tensors
     new_tensors = write_rand_tensors(tmp_path)
     assert_tensor_list_diff(original_tensors, new_tensors)
     loaded_tensors = load_rand_tensors(tmp_path=tmp_path)
     assert_tensor_list_eq(loaded_tensors, new_tensors)
     # Update the local tensors with the original tensors from gcp
-    cfg.rsync_down(rand_folder, tmp_path, verbose=False)
+    
+    def mock_sync_down_new_file(self,remote_path: str,local_path: Path,logger: FileLogger | None = None,verbose=True):
+        for i in range(2):
+            torch.save(original_tensors[i],tmp_path.joinpath(f"t_{i}.pt"))
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.rsync_down",mock_sync_down_new_file):
+        cfg.rsync_down(rand_folder, tmp_path, verbose=False)
     loaded_tensors = load_rand_tensors(tmp_path=tmp_path)
     assert_tensor_list_eq(loaded_tensors, original_tensors)
 
     # Update a mock node from gcp
-    hostname = socket.gethostname()
+    with mock.patch("socket.gethostname", return_value="localhost"):
+        hostname = socket.gethostname()
     mock_node_path = tmp_path.joinpath(hostname).joinpath(tmp_path.name)
-    cfg.rsync_down_node(hostname, rand_folder, mock_node_path)
+    def mock_sync_down_node_new_file(self,node_hostname,remote_path: str,local_path: Path,logger: FileLogger | None = None,verbose=True):
+        mock_node_path.parent.mkdir(exist_ok=True)
+        mock_node_path.mkdir(exist_ok=True)
+        for i in range(2):
+            torch.save(original_tensors[i],mock_node_path.joinpath(f"t_{i}.pt"))
+    with mock.patch("ablator.modules.storage.cloud.GcpConfig.rsync_down_node",mock_sync_down_node_new_file):
+        cfg.rsync_down_node(hostname, rand_folder, mock_node_path)
     node_tensors = load_rand_tensors(tmp_path=mock_node_path)
     assert_tensor_list_eq(node_tensors, original_tensors)
     # TODO teardown refactoring
     cmd = ["gsutil", "-m", "rm", "-rf", rand_destination]
-    p = cfg._make_process(cmd, verbose=False)
-    out, err = p.communicate()
-    assert len(out) == 0 and len(err) > 0
+    
+    with patch("subprocess.Popen") as mock_popen:
+        mock_instance = mock_popen.return_value
```

Comment:
we would need to mock local directory rm.

Reply:
done .